Pcap++/src/MBufRawPacket.cpp

```diff
@@ -0,0 +1,332 @@
+#ifdef USE_DPDK
```

Comment:
Yes. This list summarize them:
* pointer to device is changed to pointer to mempool so packet will no be bound to device but to mempool. As in code base the devices and pools are bound thru all lifetime of device objects this will not create more issues than previous approach (like in the scenarion when MBufRawPacket have poiner for device that is already deleted - this was possible in previouse code and this is possible in same places now - there is no easy way to avoid this issue especially in C++98 where atomic reference counting is not an potion)
* added generic private methods:
```C++
bool init(struct rte_mempool* mempool);
bool initFromRawPacket(const RawPacket* rawPacket, struct rte_mempool* mempool);
```
that implements old methods but on mempool instead of device
* Methods for KNI device are added:
```C++
bool init(KniDevice* device);
bool initFromRawPacket(const RawPacket* rawPacket, KniDevice* device);
```
* Methods for DpdkDevice are rewritten in terms of generic methods
* On copy construction now the mempool is propagated not a device (strong correspondenc with documentation is saved as stated above in first list point)
Thst is all

Reply:
with `KniDevice` there is a slightly bigger concern because KNI devices may be destroyed throughout the application life cycle (as opposed to DPDK devices who are never destroyed).

But like you mentioned I'm not sure there is a good way to handle that

Reply:
I really see no use case when the KNI device may be destroyed during the application active lifetime. It most certanly will be used as a communication between Linux network stack and DPDK devices which implies lifetime simmilar to that of DPDK device (as it is currently implemented within implicit KNI device destruction on application exit).
Do You suggest to delete `DestroyDevice` method in that case?

Reply:
@seladb I want to hear Your thoughts on this topic.

Reply:
Yes, that might be a good idea. We can always consider adding this method back if someone asks to

Reply:
\+